### PR TITLE
ACMS-1984: Moderation dashboard fix for users.

### PR DESCRIPTION
**Pull request by deepakmishra2**
**Motivation**

Fixes #ACMS-1984

**Proposed changes**
Excluding view json task for users, as we dont see json data for users since 9.3

**Alternatives considered**

**Testing steps**
Follow steps from ticket

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

**Review by apathak18 — APPROVED** [review]

**Review by rajeshreeputra — APPROVED** [review]

File: modules/acquia_cms_headless/modules/acquia_cms_headless_ui/acquia_cms_headless_ui.module [hunk 1]
```diff
@@ -216,7 +215,9 @@ function acquia_cms_headless_ui_local_tasks_alter(array &$local_tasks) {
     // prefix accounts for the fact that Media module exposes the canonical
     // route conditionally based on a config switch, and therefore creates the
     // Edit and View tasks with a deriver (which is why it's prefixed).
-    elseif (preg_match('/^(media\.tasks:)?entity\.(\S*)\.canonical$/', $id, $matches)) {
+    // Pattern includes all routes with entity and ends with canonical, the API
+    // menu will be visible only on node entities to render the jsonapi data.
+    elseif (preg_match('/^(media\.tasks:)?entity\.(\S*(?<!user))\.canonical$/', $id, $matches)) {
```
Comment (chandan-singh7929) [inline modules/acquia_cms_headless/modules/acquia_cms_headless_ui/acquia_cms_headless_ui.module:220]: I don't think this statement is correct.
`the API menu will be visible only on node entities to render the jsonapi data.`

I can see it for taxonomy term, I think we should find root cause why its causing issue with user.
image

Comment (deepakmishra2) [reply modules/acquia_cms_headless/modules/acquia_cms_headless_ui/acquia_cms_headless_ui.module:220]: Even i'm not 100% sure on this solution as this looks similar to what we did to exclude users route.

Comment (chandan-singh7929) [reply modules/acquia_cms_headless/modules/acquia_cms_headless_ui/acquia_cms_headless_ui.module:220]: @apathak18 - As per discussion, please try pattern to bypass if its `views_view:view.`

File: modules/acquia_cms_common/acquia_cms_common.install [hunk 2]
```diff
@@ -304,3 +304,17 @@ function acquia_cms_common_update_8009() {
     }
   }
 }
+
+/**
+ * Implements hook_update_N().
+ *
+ * Set disabled FALSE to jsonapi resource for user to API menu.
+ */
+function acquia_cms_common_update_8010() {
```
Comment (rajeshreeputra) [inline modules/acquia_cms_common/acquia_cms_common.install:313]: update hook number for this change to 8301 for 3.x release and 8201 for 2.x release(if releasing in 2.x)

Comment (apathak18) [reply modules/acquia_cms_common/acquia_cms_common.install]: Done

File: tests/src/ExistingSite/DecoupledTest.php [hunk 3]
```diff
@@ -133,7 +133,7 @@ public function testResourceTypes(?array $roles) {
 
     // The user resource type should be disabled, so we should not be able to
     // do anything with it.
-    $this->assertResourceType(FALSE, $this->createUser(), [
```
Comment (chandan-singh7929) [inline tests/src/ExistingSite/DecoupledTest.php:136]: @apathak18 I think we disable the user resource intentionally. I think we should not enable it. Just fix the WSOD error and we are good.

@vishalkhode1 - what do you suggest.

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**